web_ui/src/components/UsageBillingPage.tsx

```diff
@@ -375,34 +426,42 @@ function ManageSubscriptionModal({
   currentSeats,
 }: IManageSubscriptionModalProps) {
   const [seats, setSeats] = React.useState(currentSeats)
-  const [loading, setLoading] = React.useState(false)
+  const [creatingSubscription, setCreatingSubscription] = React.useState(false)
   const [error, setError] = React.useState("")
   const [prorationAmount, setProrationAmount] = React.useState<
     IProrationAmount
   >({ kind: "loading" })
   const [prorationTimestamp, setProrationTimestamp] = React.useState(0)
+  const [subscriptionPeriod, setSubscriptionPeriod] = React.useState<
+    "month" | "year"
+  >(cost.planInterval)
   const seatsRef = React.useRef(0)
+  const subscriptionPeriodRef = React.useRef<"month" | "year">("month")
```

Comment:
why refs instead of useState?

Reply:
a hack for the debouncing function below